src/histograms/AbstractHistogram.ts

```diff
@@ -263,6 +255,92 @@ export abstract class AbstractHistogram {
     }
   }
 
+  public getHorizontalOffset(chartAxes){
+    let h = this.chartDimensions.height;
+    if (isChartAxes(chartAxes)) {
+      if (!this.histogramParams.yAxisFromZero) {
+        const minMax = chartAxes.yDomain.domain();
+        if (minMax[0] >= 0) {
+          h = chartAxes.yDomain(minMax[0]);
+        } else {
+          h = chartAxes.yDomain(minMax[1]);
+        }
+      } else {
+        h = chartAxes.yDomain(0);
+      }
+    }
+    return h;
+  }
+
+  public updateNumberOfLabelDisplayedIfOverlap(chartAxes: ChartAxes | SwimlaneAxes, leftOffset = 0){
+    // Get the offset used when we will draw the labels.
+    const horizontalOffset = this.getHorizontalOffset(chartAxes);
+      let sumWidth = 0;
+      // create virtual nodes. Helps to get label's width.
+      const virtualLabels = this.chartDimensions.svg.append('g');
```

Comment:
I think you can try to select the already existing labels.
Or you need to use one common method between the real label axis and the virtual one in order to avoid duplication